### PR TITLE
Clear syscall and sysregs sdbs if their files are not found

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr clears the syscall and sysregs sdbs if their files are not found, to prevent usage of sdbs that are inappropriate for the os/arch/bits/cpu.

**Test plan**

All builds should be green, including the Travis builds.

**Closing issues**

...

### CODE REVIEW

**Review by ITAYC0HEN — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Ideally, it should be something like:
```c
static bool openDatabase(Sdb **db, const char *name) {
	char *file = rz_str_newf ( RZ_JOIN_3_PATHS ("%s", RZ_SDB, "%s.sdb"),
		rz_sys_prefix (NULL), name);
	if (rz_file_exists (file)) {
		if (db && *db) {
			sdb_reset (*db);
			sdb_open (*db, file);
		} else {
			*db = sdb_new (0, file, 0);
		}
	} else {
             free (file);
             return false;
       }
       free (file);
       return true;
```
And checking the result in the place where it was called, imho.

File: test/db/analysis/avr [hunk 1]
```diff
@@ -673,36 +672,36 @@ EXPECT=<<EOF
...
-0x000001d0 1 syscall.read.2
-0x00000260 1 syscall.read.3
+0x000001d0 1 syscall.0.2
+0x00000260 1 syscall.0.3
 0x0000029e 164 fcn.0000029e
-0x000002ae 1 syscall.read.5
-0x000002f2 1 syscall.inotify_rm_watch.1
-0x00000318 1 syscall.inotify_rm_watch.0
-0x00000330 1 syscall.inotify_rm_watch
+0x000002ae 1 syscall.0.5
```
Comment (ret2libc) [inline test/db/analysis/avr:678]: I'm not sure I understand why these need to be changed. Can you explain?

Comment (XVilka) [reply test/db/analysis/avr:678]: There are no such syscalls on the AVR architecture.

**Review by wargio — APPROVED** [review]

File: test/db/analysis/avr (hunk 1 above)
Comment (ret2libc) [reply test/db/analysis/avr:678]: There are, I guess, if you have specified os=linux as it was before.
Was this test just completely wrong?

Also, are there syscalls anyway in those places?

Comment (wargio) [reply test/db/analysis/avr:678]: yes, the test was wrong.

**Comment by kazarmy** [comment]
> Ideally, it should be something like:

the suggestion lgtm

**Pull request merged**